components/cam/src/utils/error_function.F90

```diff
@@ -0,0 +1,646 @@
+module error_function
```

Comment:
What's up with this module? This is a brand new file? Where did this come from?

Reply:
It's used by the CRM 2-mom micro scheme. Since we aren't going to use the 2-mom code we don't really need it... but until we have P3 in the CRM we need the 2-mom code to work on refactoring the ECPP code, so I think we should keep it and try to remember to remove this module later.

Reply:
This code looks very similar to `cime/src/share/util/shr_spfn_mod.F90`. Is this a newer version of the existing code?